opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/RevealJsonFormFragmentPresenter.java

```diff
@@ -61,9 +64,16 @@ public ValidationStatus writeValuesAndValidate(LinearLayout mainView) {
                             onLocationValidated();
                         }
                     }
+                    break;
                 }
             }
         }
+        if (!hasMapView && validationStatus.isValid()) {
```

Comment:
Then the `hasMapView` can be done away with?

Reply:
Check above comment. Without it then its difficult to determine if a mapview is in the form and location validation was initiated if the mapview is valid .

Reply:
Cool.